File: java/src/main/java/com/openhealthhub/subscription/SubscriptionClient.java
package com.openhealthhub.subscription;

import ca.uhn.fhir.rest.api.MethodOutcome;
import ca.uhn.fhir.rest.client.api.IGenericClient;
import com.openhealthhub.util.FhirUtil;
import org.hl7.fhir.r4.model.StringType;
import org.hl7.fhir.r4.model.Subscription;

import java.util.List;

public class SubscriptionClient {

    private final IGenericClient client;

    public static void main(String... args) {
        SubscriptionClient client = new SubscriptionClient();

        FhirUtil.printResource(client.createSubscription());
    }

    SubscriptionClient() {
        client = FhirUtil.createClient();
    }

    MethodOutcome createSubscription() {
        Subscription.SubscriptionChannelComponent channel = new Subscription.SubscriptionChannelComponent()
                .setType(Subscription.SubscriptionChannelType.RESTHOOK)
                .setHeader(List.of(new StringType("Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")))
                .setEndpoint("https://your-webhook/endpoint");
        Subscription subscription = new Subscription()
                .setCriteria("QuestionnaireResponse?author=johndoe")
                .setStatus(Subscription.SubscriptionStatus.REQUESTED)
                .setChannel(channel);

        return client.create()
                .resource(subscription)
                .execute();

    }
}

File: php/src/OpenHealthhub/PhpFhirClient/SubscriptionClient.php
<?php

namespace OpenHealthhub\PhpFhirClient;


use DCarbone\PHPFHIRGenerated\R4\FHIRElement\FHIRBackboneElement\FHIRSubscription\FHIRSubscriptionChannel;
use DCarbone\PHPFHIRGenerated\R4\FHIRElement\FHIRSubscriptionChannelType;
use DCarbone\PHPFHIRGenerated\R4\FHIRElement\FHIRSubscriptionStatus;
use DCarbone\PHPFHIRGenerated\R4\FHIRResource\FHIRDomainResource\FHIRSubscription;

class SubscriptionClient
{
    public function createSubscription(): FHIRSubscription
    {
        $client = new FhirClient();
        $subscription = new FHIRSubscription();
        $channel = new FHIRSubscriptionChannel();
        $channelType = new FHIRSubscriptionChannelType();
        $subscription->setCriteria('QuestionnaireResponse?author=johndoe')->setChannel($channel->setEndpoint('https://your-webhook/endpoint')->setType($channelType->setValue('rest-hook'))->addHeader(['Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8']));
        $status = new FHIRSubscriptionStatus(['requested']);
        $subscription->setStatus($status);
        $res = $client->create('Subscription', $subscription);
        return new FHIRSubscription($res);
    }
}
